scripts/qa_crowbarsetup.sh

```diff
@@ -3006,6 +3031,43 @@ function addfloatingip
     fi
 }
 
+function ha_services_failover_test
+{
+    #Get cluster name
+    local cluster=`get_cluster_name`
+    read -a nodes <<< `get_all_nodes_from_cluster "$cluster"`
+    #ssh to first node kill services nova-api ,glance-api,pkill mysql
+    ssh ${nodes[0]} 'pkill nova-api; pkill glance-api; pkill mysql'
+    #Start rally tests
+    run_rally_test
+    #ssh to nodes check that services running
+    for((i=1; i<${#nodes[@]}; i++)) ;do
+        ssh ${nodes[$i]} 'pgrep nova-api; pgrep glance-api; pgrep mysql'
+        echo $?
+    done
+    #Check that services recovered
+    ssh ${nodes[0]} 'pgrep nova-api; pgrep glance-api; pgrep mysql'
+    local ret=$?
+    if [ "$ret" = "0" ]; then
+        echo "Services recovered succesfully"
+    else
+        echo "Services fail to recover"
+    fi
+}
+
+function run_rally_test
+{
+    #Get network id
+    get_novacontroller
+    local net_id=`oncontroller get_network_id`
+    . scripts/jenkins-support.sh
+    #Pass parameters to rally server
+    echo -e "cloud=$cloud \ntask=$task \nnetid=$net_id \nimage_name=$image_name" > setenv
```

Comment:
Where do you set `task`? I can't find it :slightly_frowning_face:.

Reply:
fixed